### PR TITLE
diffie-hellman: add immutability / purity test case

### PR DESCRIPTION
**Pull request by SleeplessByte**
JavaScript has a test case to test if the algorithm is side-effect free, or has the appearance that it is. 

- Tracks that have a functional implementation may test if running the same function twice in succession generates the same result (not this test case)
- Tracks that have an object-oriented approach may test if running the same method with different input generates the correct result (which it arguably wouldn't if p or g were mutated in the process)

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/diffie-hellman/canonical-data.json [hunk 1]
```diff
@@ -45,6 +47,26 @@
...
+        "public key from different values, for example from the test with     ",
+        "uuid (b4161d8e-53a1-4241-ae8f-48cc86527f22).                         "
+      ],
+      "property": "publicKey",
+      "input": {
+        "p": 23,
+        "g": 5,
+        "privateKey": 15
+      },
+      "expected": 19,
```
Comment (petertseng) [inline exercises/diffie-hellman/canonical-data.json:65]: this value is correct because `5 ** 15 % 23 == 19`.

**Review by petertseng — APPROVED** [review]
the case added is correct. please do not squash commits when merging. the schema change would ideally go into its own commit. I express no views on other issues such as immutability.

**Comment by SleeplessByte** [comment]
For the person merging: Do **not** squash.

**Comment by ErikSchierboom** [comment]
@SleeplessByte There is a merge conflict.

**Comment by SleeplessByte** [comment]
@exercism/reviewers it's ready!

**Review by neenjaw — APPROVED** [review]
🎉

**Comment by SleeplessByte** [comment]
Ha. I swear I didn't press squash 🤷🏽

**Pull request merged**